bin/sozo/src/utils.rs

```diff
@@ -0,0 +1,33 @@
+use anyhow::Result;
+use dojo_world::utils::{execution_status_from_maybe_pending_receipt, TransactionWaiter};
+use starknet::core::types::{ExecutionResult, InvokeTransactionResult};
+use starknet::providers::Provider;
+
+pub async fn handle_transaction_result<P>(
+    provider: P,
+    transaction_result: InvokeTransactionResult,
+    wait_for_tx: bool,
+    show_receipt: bool,
+) -> Result<()>
+where
+    P: Provider + Send,
+{
+    if wait_for_tx {
+        let receipt =
+            TransactionWaiter::new(transaction_result.transaction_hash, &provider).await?;
+
+        if show_receipt {
+            println!("{}", serde_json::to_string_pretty(&receipt)?);
+        } else {
+            let status = match execution_status_from_maybe_pending_receipt(&receipt) {
+                ExecutionResult::Succeeded => "Transaction OK",
+                ExecutionResult::Reverted { reason: _ } => "Transaction REVERTED",
+            };
+            println!("{status} (hash: {:#x})", transaction_result.transaction_hash);
```

Comment:
Should we print the status even if it's succeeded? Imo, it's seems more natural to only print this if the tx failed.

Reply:
and can we keep the same format as without the `--wait` flag to display the hash. Makes it more consistent this way. 

```
Transaction hash: 0x123124 (Reverted with reason: invalid signature)
```
```
Transaction hash: 0x123124 (Ok)
```

```
Transaction hash: 0x123124
Status: REVERTED (Invalid signature)
```

```
Transaction hash: 0x123124
Status: OK
```

just giving some suggestions on the print format

Reply:
Good call @kariy. I would agree with your last sentence, and say (1) is cleaner and easier to read yes.